go-controller/pkg/network-attach-def-controller/network_attach_def_controller_test.go

```diff
@@ -434,3 +431,124 @@ func TestNetAttachDefinitionController(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncAll(t *testing.T) {
+	network_A := &ovncnitypes.NetConf{
+		Topology: types.Layer2Topology,
+		NetConf: cnitypes.NetConf{
+			Name: "network_A",
+			Type: "ovn-k8s-cni-overlay",
+		},
+		MTU: 1400,
+	}
+	network_B := &ovncnitypes.NetConf{
+		Topology: types.LocalnetTopology,
+		NetConf: cnitypes.NetConf{
+			Name: "network_B",
+			Type: "ovn-k8s-cni-overlay",
+		},
+		MTU: 1400,
+	}
+	type TestNAD struct {
+		name    string
+		netconf *ovncnitypes.NetConf
+	}
+	tests := []struct {
+		name     string
+		testNADs []TestNAD
+	}{
+		{
+			name: "multiple networks referenced by multple nads",
+			testNADs: []TestNAD{
+				{
+					name:    "test/nad1",
+					netconf: network_A,
+				},
+				{
+					name:    "test/nad2",
+					netconf: network_B,
+				},
+				{
+					name:    "test/nad3",
+					netconf: network_A,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		g := gomega.NewWithT(t)
+
+		config.OVNKubernetesFeature.EnableMultiNetwork = true
+		fakeClient := util.GetOVNClientset().GetOVNKubeControllerClientset()
+		wf, err := factory.NewOVNKubeControllerWatchFactory(fakeClient)
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+
+		tncm := &testNetworkControllerManager{
+			controllers: map[string]NetworkController{},
+		}
+		nadController, err := NewNetAttachDefinitionController(
+			"SUT",
+			tncm,
+			wf,
+		)
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+
+		expectedNetworks := map[string]util.NetInfo{}
+		for _, testNAD := range tt.testNADs {
+			namespace, name, err := cache.SplitMetaNamespaceKey(testNAD.name)
+			g.Expect(err).ToNot(gomega.HaveOccurred())
+			testNAD.netconf.NADName = testNAD.name
+			nad, err := buildNAD(name, namespace, testNAD.netconf)
+			g.Expect(err).ToNot(gomega.HaveOccurred())
+			_, err = fakeClient.NetworkAttchDefClient.K8sCniCncfIoV1().NetworkAttachmentDefinitions(namespace).Create(
+				context.Background(),
+				nad,
+				v1.CreateOptions{},
+			)
+			g.Expect(err).ToNot(gomega.HaveOccurred())
+			netInfo := expectedNetworks[testNAD.netconf.Name]
+			if netInfo == nil {
+				netInfo, err = util.NewNetInfo(testNAD.netconf)
+				g.Expect(err).ToNot(gomega.HaveOccurred())
+				expectedNetworks[testNAD.netconf.Name] = netInfo
+			}
+		}
+
+		err = wf.Start()
+		g.Expect(err).ToNot(gomega.HaveOccurred())
+		defer wf.Shutdown()
```

Comment:
my IDE caught this: you call defer in a `for _, tt := range tests` loop, but defer will only be executed at the end of the function.

Reply:
hrrrrmmm my test generator did a bad job here, each test is supposed to be called in its own function